pkg/evaluator/builtin.go

```diff
@@ -376,7 +376,8 @@ func str2numFunc(scope *scope, args []Value) (Value, error) {
 	s := args[0].(*String)
 	n, err := strconv.ParseFloat(s.Val, 64)
 	if err != nil {
-		setGlobalErr(scope, "str2num: cannot parse "+s.Val)
+		msg := fmt.Sprintf("str2num: cannot parse %q", s.Val)
```

Comment:
these changes would probably belong best when you add the conversion section, but it doesn't matter much to me. Just pointing it out in case that was what you intended.

Reply:
conversion section comes in the next PR i think, but I also just want to tick it off now, so if it's ok I will leave it in here.